### PR TITLE
Handle already loaded image

### PR DESCRIPTION
**Pull request by hitmands**
further information: http://stackoverflow.com/questions/12354865/image-onload-event-and-browser-cache

### CODE REVIEW

File: src/lazyload.js [hunk 1]
```diff
@@ -179,9 +179,9 @@ class LazyLoad {
         }
 
         if (element.tagName === "IMG" || element.tagName === "IFRAME") {
-            element.addEventListener("load", loadCallback);
             element.addEventListener("error", errorCallback);
             element.classList.add(settings.class_loading);
+            element.complete ? loadCallback.call(element) : element.addEventListener("load", loadCallback);
```
Comment (verlok) [inline src/lazyload.js]: This line would cause an error when the element is an IFRAME

**Comment by verlok** [comment]
Also, this change is not necessary since everything is working on latest version. See the "callbacks.html" and "fade_in" demos.

**Comment by hitmands** [comment]
The `complete` property doesn't exist in the [`HTMLIframeElement`](https://developer.mozilla.org/it/docs/Web/API/HTMLIFrameElement) but this will not cause any error, the `HTMLIframeElement` inherits from `Object` and any `undefined object property` is interpreted as a [`falsy value`](https://developer.mozilla.org/it/docs/Glossary/Falsy) in javascript (which means "the behaviour when `element instanceof HTMLIframeElement` unchanges).

The change should be considered because represents a **performance improvement** since it checks if the image is already loaded and then fires the `loadCallback` without any delay.

The bug is also verifiable in any `webkit` implementation:  https://bugs.chromium.org/p/chromium/issues/detail?id=7731

**Comment by verlok** [comment]
You would be right if the `complete` check would be after the `src` is set, but the point you're willing to edit is _before_ then. 
Furthermore, if the user sets a "loading" image in the `src` and the _lazy_ image in the `data-src` attribute, the `complete` property would be true even before the lazy load had started.

**Comment by hitmands** [comment]
Ok, the change should take care about the placeholder image too, 
but, the point is: **when the image is already cached** the `loadCallback` won't be fired in some `webkit` implementations. We attached many relatives resourced. We think that should be considered. I'm going to close the pr.

**Comment by verlok** [comment]
Thanks, I'll take a look at that, as in the issue opened by @lucavolta

**Pull request closed without merging**